rust_src/src/fonts.rs

```diff
@@ -97,6 +103,74 @@ impl FontExtraType {
     }
 }
 
+pub type LispFontObjectRef = ExternalPtr<Lisp_Font_Object>;
+
+impl LispFontObjectRef {
+    pub fn add_log(self, action: &str, result: LispObject) {
+        let c_str = CString::new(action).unwrap();
+        unsafe { font_add_log(c_str.as_ptr(), self.into(), result) }
+    }
+
+    pub fn close(mut self, mut _frame: LispFrameRef) {
+        if data::aref(self.into(), FONT_TYPE_INDEX.into()).is_nil() {
+            // Already closed
+            return;
+        }
+        self.add_log("close", LispObject::from(false));
+        unsafe {
+            if let Some(f) = (*self.driver).close {
+                f(self.as_mut())
+            }
+            #[cfg(feature = "window-system")]
+            {
+                #[cfg(feature = "x11")]
+                let mut display_info = &mut *(*_frame.output_data.x).display_info;
+                #[cfg(feature = "nextstep")]
+                let mut display_info = &mut *(*_frame.output_data.ns).display_info;
+                #[cfg(feature = "w32")]
+                let mut display_info = &mut *(*_frame.output_data.w32).display_info;
+                debug_assert!(display_info.n_fonts > 0);
+                display_info.n_fonts -= 1;
+            }
+        }
+    }
+}
+
+impl From<LispFontObjectRef> for LispObject {
+    fn from(f: LispFontObjectRef) -> Self {
+        LispObject::tag_ptr(f, Lisp_Type::Lisp_Vectorlike)
+    }
+}
+
+impl From<LispObject> for LispFontObjectRef {
+    fn from(o: LispObject) -> Self {
+        o.as_font_object_or_error()
```

Comment:
@nickdrozd has been working to remove these methods and replace them with From conversions. Fold the method code here and drop it.

Reply:
Yeah, I noticed the pr's about that. done.